edu_edfi_airflow/dags/earthbeam_dag.py

```diff
@@ -78,6 +79,81 @@ def build_local_raw_dir(self, tenant_code: str, api_year: int, grain_update: Opt
         )
         return raw_dir
 
+    @staticmethod
+    def partition_on_tenant_and_year(
+        csv_paths: Union[Union[str, Path], List[Union[str, Path]]],
+        output_dir: str,
+        tenant_col: str = "tenant_code",
+        tenant_map: dict = None,
+        year_col: str = "api_year",
+        year_map: dict = None,
+    ):
+        """
+        Preprocessing function to shard data to parquet on disk
+
+        :param csv_paths: one or more complete file paths pointing to input data
+        :param output_dir: root directory of the parquet
+        :param tenant_col: (optional) name of the column to use as tenant code
+        :param tenant_map: (optional) map values from the contents of tenant_col to valid tenant codes
+        :param year_col: (optional) name of the column to use as API year
+        :param year_map: (optional) map values from the contents of api_col to valid API years
+
+        :return:
+        """
+        # (expensive) imports here so that the airflow scheduler doesn't have to deal with them
+        import dask
+        import dask.dataframe as dd
+        import pandas as pd
+
+        tenant_code = "tenant_code"
+        api_year = "api_year"
+
+        input_is_list = True if isinstance(csv_paths, list) else False
+
+        final_csv_paths = []
+        if input_is_list:
+            final_csv_paths = csv_paths
+        else:
+            # otherwise we have a string, so wrap it in a list
+            final_csv_paths = [csv_paths]
+
+        for csv_path in final_csv_paths:
+            if not Path(csv_path).is_file() or not Path(csv_path).suffix == '.csv':
+                raise ValueError(f"Input path '{csv_path}' is not a path to a file whose name ends with '.csv'")
```

Comment:
Is there a world where we want to be flexible to filetype and accept others than CSV? I'm legitimately asking here; I'm not sure.

Reply:
IMO there probably is such a world, and when it dawns, we can update this function